packages/protocol-http/src/Fields.ts

```diff
@@ -0,0 +1,42 @@
+import { Field } from "./Field";
+import { FieldPosition } from "./FieldPosition";
+
+export type FieldsOptions = { fields?: Field[]; encoding?: string; };
+
+/**
+ * Collection of Field entries mapped by name.
+ */
+export class Fields {
+  private readonly entries: Record<string, Field> = {};
+  private readonly encoding: string;
+
+  constructor(options: FieldsOptions) {
```

Comment:
this could be

```js
public constructor({ fields = [], encoding = "utf-8" }: FieldsOptions) {
  fields.forEach(this.setField.bind(this));
  this.encoding = encoding;  
}
```